packages/frontend/src/components/UserSettingsModal/CreateTokenPanel/index.tsx

```diff
@@ -0,0 +1,153 @@
+import { Button, Card, InputGroup, Intent } from '@blueprintjs/core';
+import { CreatePersonalAccessToken, formatTimestamp } from '@lightdash/common';
+import React, { FC, useEffect, useState } from 'react';
+import CopyToClipboard from 'react-copy-to-clipboard';
+import { useForm } from 'react-hook-form';
+import { useCreateAccessToken } from '../../../hooks/useAccessToken';
+import { useApp } from '../../../providers/AppProvider';
+import {
+    AccessTokenForm,
+    BackButton,
+    ExpireDateSelect,
+    InvitedCallout,
+    InviteFormGroup,
+    Panel,
+    ShareLinkCallout,
+    SubmitButton,
+    TokeDescriptionInput,
+} from './CreateTokens.styles';
+
+const CreateTokenPanel: FC<{
+    onBackClick: () => void;
+}> = ({ onBackClick }) => {
+    const { showToastSuccess } = useApp();
+    const { data, mutate, isError, isLoading, isSuccess } =
+        useCreateAccessToken();
+
+    const [expireDate, setExpireDate] = useState<Date | undefined>();
+    const currentDate = new Date();
+    const methods = useForm<CreatePersonalAccessToken>({
+        mode: 'onSubmit',
+    });
+
+    useEffect(() => {
+        if (isError) {
+            methods.reset({ ...methods.getValues() }, { keepValues: true });
+        }
+        if (isSuccess) {
+        }
+    }, [isError, methods, isSuccess]);
+
+    const handleSubmit = (formData: CreatePersonalAccessToken) => {
+        const expiredValue = !!Number(formData?.expiresAt)
+            ? new Date(
+                  currentDate.setDate(
+                      currentDate.getDate() + Number(formData?.expiresAt),
+                  ),
+              )
```

Comment:
The `currentDate` is initialised outside this function so it could cause weird bugs. Eg: I have this page open for 2 days and then click the submit button. The current code is going to use the  date from 2 days ago.